src/factory.ts

```diff
@@ -0,0 +1,168 @@
+     * @remarks In utility functions other than {@link defineConfig}, this is automatically generated.
+     */
+    side: typeof sides[number];
+    /**
+     * A list of extra "external" dependencies, (i.e., dependencies that the
+     * {@link side}'s runtime already provides and thus doesn't need to be bundled)
+     */
+    extraExternal?: string[];
+    /**
+     * A list of extra ESBuild {@link Plugin}s to be applied after the default ones.
+     */
+    extraPlugins?: Plugin[],
+    /**
+     * Extra {@link BuildOptions} to append to the finalized ESBuild config.
+     * Alternatively, you can destructure {@link defineConfig} and override default options.
+     */
+    extraConfig?: BuildOptions,
+    /**
+     * Whether or not to compile the extension's source code to ESM.
+     */
+    esm?: boolean;
+}
+
+export function defineConfig({
+  name,
+  entry,
+  output,
+  side,
+  extraExternal = [],
+  extraPlugins = [],
+  extraConfig = {},
+  esm = false
+}: ESBuildFactoryOptions): BuildOptions | null {
+  const entryPoints: Array<{ in: string; out: string } | string> = [];
+  if (side !== "webpackModules") {
+    for (const fileExt of fileExts) {
+      const filePath = path.join(entry, `${side}.${fileExt}`);
+      if (fs.existsSync(filePath)) entryPoints.push(filePath);
+    }
+  }
+
+  const wpModulesDir = path.join(entry, "webpackModules");
+  if (side === "webpackModules" && fs.existsSync(wpModulesDir)) {
+    const wpModules = fs.readdirSync(wpModulesDir);
+    for (const wpModule of wpModules) {
+      if (fs.statSync(path.join(wpModulesDir, wpModule)).isDirectory()) {
+        for (const fileExt of fileExts) {
+          const filePath = path.join(
+            wpModulesDir,
+            wpModule,
+            `index.${fileExt}`
+          );
+          if (fs.existsSync(filePath))
+            entryPoints.push({
+              in: filePath,
+              out: wpModule
+            });
+        }
+      } else {
+        entryPoints.push(path.join(wpModulesDir, wpModule));
+      }
+    }
+  }
+
+  if (entryPoints.length === 0) return null;
+
+  return {
+    entryPoints: entryPoints as BuildOptions['entryPoints'],
+    outdir: side === "webpackModules" ? path.join(output, "webpackModules") : output,
+
+    format: esm && side === "index" ? "esm" : "iife",
+    globalName: "module.exports",
+    platform: ["index", "webpackModules"].includes(side) ? "browser" : "node",
+
+    treeShaking: true,
+    bundle: true,
+    minify: prod,
+    sourcemap: "inline",
+
+    external: [...external, ...extraExternal],
+    plugins: [
+      copyFile(path.join(entry, "manifest.json"), path.join(output, "manifest.json")),
+      copyFile(path.join(entry, "style.css"), path.join(output, "style.css")),
+      webpackImports,
+      betterLogging(`${name}/${side}`),
+      ...extraPlugins
+    ],
+
+    ...extraConfig
+  };
+}
+
+export function defineConfigs(options: Omit<ESBuildFactoryOptions, 'side'>): BuildOptions[] {
```

Comment:
This seems inconsistent with formatting, I'm pretty sure our Prettier config uses double quotes. Can you make sure these files have been fully formatted?